### PR TITLE
refactor: pull column filling logic out of mito worker loop

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

As title says. Also with some other changes that reduce some overhead.

We can further pull more logic out of the worker, like primary key encoding or row to column conversion.

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: src/mito2/src/request.rs [hunk 1]
```diff
@@ -875,7 +917,7 @@ mod tests {
             rows: vec![],
         };
 
-        let err = WriteRequest::new(RegionId::new(1, 1), OpType::Put, rows).unwrap_err();
```
Comment (evenyag) [inline src/mito2/src/request.rs:878]: nitpick: Add a unit test that the metadata is not None.

File: src/mito2/src/request.rs [hunk 2]
```diff
@@ -248,46 +256,67 @@ impl WriteRequest {
...
+            if default_value.value_data.is_some() {
+                default_values.push(default_value);
+                columns_to_fill.push(column);
+            }
         }
 
-        // Insert default value to each row.
         for row in &mut self.rows.rows {
-            row.values.push(proto_value.clone());
+            row.values.extend(default_values.clone());
```
Comment (evenyag) [inline src/mito2/src/request.rs:299]: non-blocking: Maybe we can only clone the items.
```suggestion
            row.values.extend(default_values.iter().cloned());
```

File: src/mito2/src/request.rs [hunk 3]
```diff
@@ -66,13 +66,20 @@ pub struct WriteRequest {
     has_null: Vec<bool>,
     /// Write hint.
     pub hint: Option<WriteHint>,
+    /// Region metadata on the time of this request is created.
+    pub region_metadata: Option<RegionMetadataRef>,
```
Comment (evenyag) [inline src/mito2/src/request.rs:70]: non-blocking: maybe `pub(crate)`.

**Review by evenyag — APPROVED** [review]
Others LGTM

**Review by WenyXu — APPROVED** [review]
LGTM
